src/test/java/org/example/Operations/CollectionOperationsTest.java

```diff
@@ -1,250 +1,483 @@
+        alice.authenticate();
+        aliceToken = alice.getAuthToken();
+
+        resetPath(rods, token, "/tempZone/home/rods");
+    }
+
+    @Test
+    public void testCommonOperations() {
+        String collectionPath = "/tempZone/home/rods/common_ops";
+
+        // Create a new Collection.
+        res = rods.collections().create(token, collectionPath, OptionalInt.empty());
+        assertEquals("Creating collection request failed", 200, res.getHttpStatusCode());
+        assertEquals("Creating collection failed", 0,
+                getIrodsResponseStatusCode(res.getBody()));
+
+        // Stat the collection to show that it exists.
+        res = rods.collections().stat(token, collectionPath, Optional.empty());
+        assertEquals("Stat request failed", 200, res.getHttpStatusCode());
+        assertEquals("Stat failed", 0,
+                getIrodsResponseStatusCode(res.getBody()));
+
+        // Rename the collection.
+        String newCollectionPath = collectionPath + ".renamed";
+        res = rods.collections().rename(token, collectionPath, newCollectionPath);
+        assertEquals("Renaming collections request failed", 200, res.getHttpStatusCode());
+        assertEquals("Renaming collections failed", 0,
+                getIrodsResponseStatusCode(res.getBody()));
+
+        // Stat the original collection to show that it does not exist.
+        res = rods.collections().stat(token, collectionPath, Optional.empty());
+        assertEquals("Stat request failed", 200, res.getHttpStatusCode());
+        assertEquals("Stat did not fail as expected", -170000,
+                getIrodsResponseStatusCode(res.getBody()));
+        // -170000: NOT_A_COLLECTION
+
+        // Stat the new collection to show that it does exist.
+        res = rods.collections().stat(token, newCollectionPath, Optional.empty());
+        assertEquals("Stat request failed", 200, res.getHttpStatusCode());
+        assertEquals("Stat failed", 0,
+                getIrodsResponseStatusCode(res.getBody()));
+
+        // Give another user (alice) permission to read the object.
+        res = rods.collections().set_permission(token, newCollectionPath, "alice", Permission.READ,
+                OptionalInt.empty());
+        assertEquals("Setting permission request failed", 200, res.getHttpStatusCode());
+        assertEquals("Setting permission failed", 0,
+                getIrodsResponseStatusCode(res.getBody()));
+
+        // Show that the alice user now has read permission on the collection.
+        res = rods.collections().stat(token, newCollectionPath, Optional.empty());
+        assertEquals("Stat request failed", 200, res.getHttpStatusCode());
+        assertEquals("Stat failed", 0,
+                getIrodsResponseStatusCode(res.getBody()));
+
+        JsonNode rootNode;
+        try {
+            rootNode = mapper.readTree(res.getBody());
+        } catch (JsonProcessingException e) {
+            throw new RuntimeException(e);
+        }
```

Comment:
JUnit should offer an assertion for handling things that can throw. Prefer using that rather to try-catch.

Reply:
I've replaced all the try-catches